__tests__/resolveConfig.test.js

```diff
@@ -430,3 +430,228 @@ test('functions in the user theme section are lazily evaluated', () => {
     },
   })
 })
+
+test('theme values in the extend section extend the existing theme', () => {
+  const userConfig = {
+    theme: {
+      extend: {
+        opacity: {
+          '25': '25',
+          '75': '.75',
+        },
+        backgroundColors: {
+          customBackground: '#bada55',
+        },
+      },
+    },
+  }
+
+  const defaultConfig = {
+    prefix: '-',
+    important: false,
+    separator: ':',
+    theme: {
+      colors: {
+        cyan: 'cyan',
+        magenta: 'magenta',
+        yellow: 'yellow',
+      },
+      opacity: {
+        '0': '0',
+        '50': '.5',
+        '100': '1',
+      },
+      backgroundColors: ({ colors }) => colors,
+    },
+    variants: {
+      backgroundColors: ['responsive', 'hover', 'focus'],
+      opacity: ['responsive', 'hover', 'focus'],
+    },
+  }
+
+  const result = resolveConfig([userConfig, defaultConfig])
+
+  expect(result).toEqual({
+    prefix: '-',
+    important: false,
+    separator: ':',
+    theme: {
+      colors: {
+        cyan: 'cyan',
+        magenta: 'magenta',
+        yellow: 'yellow',
+      },
+      opacity: {
+        '0': '0',
+        '50': '.5',
+        '100': '1',
+        '25': '25',
+        '75': '.75',
+      },
+      backgroundColors: {
+        cyan: 'cyan',
+        magenta: 'magenta',
+        yellow: 'yellow',
+        customBackground: '#bada55',
+      },
+    },
+    variants: {
+      backgroundColors: ['responsive', 'hover', 'focus'],
+      opacity: ['responsive', 'hover', 'focus'],
+    },
+  })
+})
+
+test('theme values in the extend section extend the user theme', () => {
+  const userConfig = {
+    theme: {
+      opacity: {
+        '0': '0',
+        '20': '.2',
+        '40': '.4',
+      },
+      height: theme => theme.width,
```

Comment:
Are you intentionally setting the `theme.width` as the `height` value?

Reply:
Yeah, test is to verify that if you have a closure value in your user config that you can still extend that same value and have the extensions add to the user overrides 👍